src/parser/Parser.cpp

```diff
@@ -1,122 +1,109 @@
 #include <Parser.h>
 
-Parser::Parser(Lexer lexer)
-    : m_lexer(std::move(lexer))
-    , m_currentToken(m_lexer.getToken())
+Parser::Parser(const Lexer lexer)
+    : m_lexer(lexer)
 {
 
 }
 
-void Parser::buildTree() {
-    parseStatement();
+std::unique_ptr<AST::Stmt> Parser::parse() {
+    return parseStmt();
 }
 
-std::unique_ptr<AST::Stmt> Parser::parseStatement() {
-    std::unique_ptr<AST::Stmt> statement;
-    Token identifier;
-    switch(m_currentToken.type) {
-        case TokenType::Def:
-            statement = parseFunction();
-        case TokenType::Return:
-            statement = parseReturn();
-        case TokenType::Identifier:
-            identifier = m_currentToken;
-            getNextToken();
-            switch(m_currentToken.type) {
-                case TokenType::LeftParen:
-                    getNextToken();
-            }
-            break;
-
-        // todo: other statements
-
-        default:
-            statement = parseExpression();
+std::unique_ptr<AST::Stmt> Parser::parseStmt(){
+    Token currTok = m_lexer.getToken();
+    if (currTok.type == TokenType::EOL || currTok.type == TokenType::END) {
```

Comment:
What is `TokenType::END`? Is it right braces or something else?

Reply:
I think it's the end of the file: [relevant](https://github.com/ubclaunchpad/durian/blob/master/src/lexer/Lexer.cpp#L13)

```cpp
Token Lexer::getToken() {
    std::string literal;
    if (m_iter == m_input.cend()) return Token {TokenType::END, m_line, ""};
    while (*m_iter == ' ') {
        ++m_iter;
    }
    ...
```

Reply:
Yes, it’s for the end-of-file. Originally I named it EOF but apparently the standard library defines that name as a macro that expands out to -1, which made the compiler deeply unhappy. So I picked END instead. I can add a comment to TokenType explaining that later.

Reply:
Right braces should be the RightBrace TokenType.